pkg/expr/dataplane.go

```diff
@@ -0,0 +1,118 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/grafana/dataplane/sdata"
+	"github.com/grafana/dataplane/sdata/numeric"
+	"github.com/grafana/dataplane/sdata/reader"
+	"github.com/grafana/dataplane/sdata/timeseries"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/expr/mathexp"
+)
+
+func shouldUseDataplane(frames data.Frames) (k data.FrameTypeKind, b bool, e error) {
+	k = data.KindUnknown
+	if len(frames) == 0 {
+		return data.KindUnknown, true, nil
+	}
+
+	firstFrame := frames[0]
+	if firstFrame == nil {
+		return
+	}
+
+	if firstFrame.Meta == nil {
+		return
+	}
+
+	if firstFrame.Meta.Type == "" {
+		return
+	}
+
+	if firstFrame.Meta.TypeVersion.IsZero() {
+		return
+	}
```

Comment:
That function does check these things (except vZero, which comes back as version warning), so since in this case I know I wanted to fall out of the dataplane code path, doing it here as well so as not to deal with specific errors.